reporter/reporter.go

```diff
@@ -73,6 +74,8 @@ func Report(ctx context.Context, err error) error {
 
 	if r, ok := FromContext(ctx); ok {
 		return r.Report(ctx, e)
+	} else {
+		return errors.New("No reporter in provided context.")
```

Comment:
Should we just `panic` here instead? This would be mostly a configuration issue if it gets hit and my guess is most consumers of reporter.Report don't check the error.